### PR TITLE
feat(profiles): Publish a tar.gz for profile directories

### PR DESCRIPTION
**Pull request by lwander**
@jtk54 what's the best way for me to run/validate this?

### CODE REVIEW

File: dev/generate_bom.py [hunk 1]
```diff
@@ -277,7 +277,7 @@ def publish_bom(self, bom_path):
     result = run_quick('hal admin publish bom --color false --bom-path {0}'
                        .format(bom_path))
     if result.returncode != 0:
-      print "'hal admin publish bom' command failed with: \n{0}\n exiting...".format(result.stdout)
+      print "'hal admin publish bom' command failed with: \n{0}\nexiting...".format(result.stderr)
```
Comment (jtk54) [inline dev/generate_bom.py]: @ewiseblatt correct me if I'm wrong, isn't `stderr` piped through `stdout` in `run_quick`?

File: dev/generate_bom.py [hunk 2]
```diff
@@ -289,14 +289,25 @@ def __publish_config(self, component, profile_path):
     """
     for profile in os.listdir(profile_path):
       full_profile = os.path.join(profile_path, profile)
+      profile_path = full_profile
```
Comment (jtk54) [inline dev/generate_bom.py]: I think `profile_path` and `full_profile` will always point to the individual profile files in `halconfig` the way this is invoked now. Seems like you can get rid of the loop and just tar up the `profile_path`, which is a directory.

Comment (jtk54) [inline dev/generate_bom.py]: Talked offline, turns out there can be directories in the halconfig `profile_path`, which we'd want to tar up in the future, while treating the toplevel files the same.

File: dev/generate_bom.py (hunk 1 above)
Comment (lwander) [inline dev/generate_bom.py]: Yup looks like that's right

Comment (ewiseblatt) [inline dev/generate_bom.py]: your right.
we can split them with a kwarg if you need.

**Pull request merged**